File: workfloworchestrator/src/main/java/com/greenops/workfloworchestrator/ingest/dbclient/DbKey.java
package com.greenops.workfloworchestrator.ingest.dbclient;

import org.apache.logging.log4j.util.Strings;

import java.util.List;

public class DbKey {

    public static String makeDbTeamKey(String orgName, String teamName) {
        return orgName + "-" + teamName;
    }

    public static String makeDbStepKey(String orgName, String teamName, String pipelineName, String stepName) {
        return Strings.join(List.of(orgName, teamName, pipelineName, stepName), '-');
    }

    public static String makeClientRequestQueueKey(String orgName, String clusterName) {
        return Strings.join(List.of(orgName, clusterName, "events"), '-');
    }

    public static String makeDbMetadataKey(String orgName, String teamName, String pipelineName, String stepName) {
        return Strings.join(List.of(orgName, teamName, pipelineName, stepName, "meta"), '-');
    }

    public static String makeDbPipelineInfoKey(String orgName, String teamName, String pipelineName) {
        return Strings.join(List.of(orgName, teamName, pipelineName, "meta"), '-');
    }

    public static String makeDbListOfTeamsKey(String orgName) {
        return orgName + "-teams";
    }

    public static String makeDbListOfStepsKey(String orgName, String teamName, String pipelineName) {
        return orgName + "-" + teamName + "-" + pipelineName + "-step";
    }

    public static String makeDbClusterKey(String orgName, String clusterName) {
        return orgName + "-" + clusterName;
    }
}

File: workfloworchestratorgo/ingest/dbkey/dbkey.go
package dbkey

import "strings"

func MakeDbTeamKey(orgName, teamName string) string {
	return orgName + "-" + teamName
}

func MakeDbStepKey(orgName, teamName, pipelineName, stepName string) string {
	return strings.Join([]string{orgName, teamName, pipelineName, stepName}, "-")
}

func MakeClientRequestQueueKey(orgName, clusterName string) string {
	return strings.Join([]string{orgName, clusterName, "events"}, "-")
}

func MakeDbMetadataKey(orgName, teamName, pipelineName, stepName string) string {
	return strings.Join([]string{orgName, teamName, pipelineName, stepName, "meta"}, "-")
}

func MakeDbPipelineInfoKey(orgName, teamName, pipelineName string) string {
	return strings.Join([]string{orgName, teamName, pipelineName, "meta"}, "-")
}

func MakeDbListOfTeamsKey(orgName string) string {
	return orgName + "-teams"
}

func MakeDbClusterKey(orgName, clusterName string) string {
	return orgName + "-" + clusterName
}
